config.py

```diff
@@ -99,30 +101,55 @@ def setup_logging(logfile: str):
     logging.info("Configuration file loading...")
 
 
-def setup(filename: str) -> None:
-    """
-    Sets up the module by loading the specified configuration file from disk
+_hash: str = ''
+
 
+def setup(filename: str) -> Dict:
+    """
+    Validate & apply configuration from disk
     Args:
         filename (str): path and filename to load.
     """
-    global config
+    # do the loading part
+    config_dict, file_hash = load_config(filename)
+    global _hash
+
+    if file_hash == _hash:
+        raise FileExistsError(f"new file's hash matches what we already have!"
```

Comment:
This is the only effective way to assert that we actually loaded a *new* configuration from disk, and it is a error state to reload the same configuration state. 

My reasoning here is the code handling the rehash itself would be expecting this exception to be raised and catch it, returning a well formatted error to the end users.

Reply:
I still find it a bit odd to use an exception for this use case. In my mind, if a function wanted to check if a value had changed, it would save a copy before and verify afterwards. Exceptions usually mean something has gone wrong. Arguably, reloading the same config should not be an error state *for this specific method*.

Reply:
given the hastles i had when initially testing this implementation, it *was* arguably an error in this method to load the same file twice.

Reply:
Seems neither you nor @shatteredbeam like the exception, so il remove it.
Rehashing will now state-agnostic.